src/domain/chainContext.ts

```diff
@@ -276,48 +320,93 @@ export class ChainContext {
       }
     });
 
-    // We run a watchdog to check if we are receiving blocks.
+    // We run a watchdog to check if we are receiving blocks. This determines if
+    // the chain is stuck or not issuing blocks. If running within a kubernetes
+    // pod, we don't exit, but we do log an error and set the sync status to unknown.
     while (true) {
       // sleep for 5 seconds
       await asyncSleep(WATCHDOG_FREQUENCY);
-      const currentTime = new Date().getTime();
-      const timeElapsed = currentTime - timeLastBlockProcessed;
+      const now = Math.floor(new Date().getTime() / 1000);
+      const timeElapsed = now - lastBlockReceived.timestamp;
 
       log.debug(`Time since last block processed: ${timeElapsed}ms`);
 
-      // If we haven't received a block in 30 seconds, exit so that the process manager can restart us
+      // If we haven't received a block within `watchdogTimeout` seconds, either signal
+      // an error or exit if not running in a kubernetes pod
       if (timeElapsed >= watchdogTimeout * 1000) {
+        const formattedElapsedTime = Math.floor(timeElapsed / 1000);
         log.error(
-          `Watchdog timeout (RPC failed, or chain is stuck / not issuing blocks)`
+          `Chain watcher last processed a block ${formattedElapsedTime}s ago (${watchdogTimeout}s timeout configured). Check the RPC.`
         );
+        if (isRunningInKubernetesPod()) {
```

Comment:
Why mentioning here Kubernetes? 
I think this could just be a setting: `exit on stale chain state`

```ts
this.sync = ChainSync.STALE;

if (exitOnStaleChainState) {
  await registry.storage.close();
  exit(1)
}
```